### PR TITLE
[ᚬpr-mirror/976] feat: wrap lock methods

### PR DESCRIPTION
**Pull request by keroro520**
* feat: Avoid locking a long time

### CODE REVIEW

File: sync/src/types.rs [hunk 1]
```diff
@@ -933,16 +921,60 @@ impl<CS: ChainStore> SyncSharedState<CS> {
 
     pub fn mark_as_known_tx(&self, hash: H256) {
         self.inflight_transactions().remove(&hash);
-        self.tx_filter().insert(hash, ());
+        self.tx_filter.lock().insert(hash, ());
     }
 
     pub fn already_known_tx(&self, hash: &H256) -> bool {
-        self.tx_filter().contains_key(hash)
+        self.tx_filter.lock().contains_key(hash)
```
Comment (driftluo) [inline sync/src/types.rs]: Use an inline function instead of a function call can reduce how much time of the lock holding? 

Maybe 1-10ms?

**Comment by doitian** [comment]
I don't recommend to submit PR to a mirror branch. That branch is force-pushed whenever PR #976 has changed, and it is deleted when 976 is closed.

File: sync/src/types.rs (hunk 1 above)
Comment (xxuejie) [reply sync/src/types.rs]: I don't really get this comment, care to explain?

Comment (keroro520) [reply sync/src/types.rs]: I just remove the `tx_filter()` method and call `tx_filter.lock()` directly. There are no differences.
So here will not reduce the lock holding time.

I am not sure I get your meaning.

Comment (driftluo) [reply sync/src/types.rs]: I seem to have misread the focus, seeing that the call becomes a function, shortening the time to acquire the lock, but in fact the focus of pr is not here.

**Comment by keroro520** [comment]
> I don't recommend to submit PR to a mirror branch. That branch is force-pushed whenever PR #976 has changed, and it is deleted when 976 is closed.

Yeah. I separate #930 into two smaller PRs, #976 and #979. And #979 dependences on the previous one.

So I simply make #979 target to #976, so that the reviewers can review the two PRs respectively.

If I don't do this, it must wait until #976 been merged.

**Pull request closed without merging**